src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.hpp

```diff
@@ -0,0 +1,286 @@
+// IWYU pragma: no_forward_declare gr::Tags::Lapse
+// IWYU pragma: no_forward_declare gr::Tags::Shift
+// IWYU pragma: no_forward_declare gr::Tags::SpacetimeMetric
+// IWYU pragma: no_forward_declare gr::Tags::SpacetimeNormalOneForm
+// IWYU pragma: no_forward_declare gr::Tags::SpacetimeNormalVector
+// IWYU pragma: no_forward_declare gr::Tags::InverseSpatialMetric
+// IWYU pragma: no_forward_declare gr::Tags::SqrtDetSpatialMetric
+
+namespace GeneralizedHarmonic {
+/*!
+ * \brief Damped harmonic gauge source function.
+ *
+ * \details See damped_harmonic_h() for details.
+ */
+template <size_t SpatialDim, typename Frame>
+struct DampedHarmonicHCompute : Tags::GaugeH<SpatialDim, Frame>,
+                                db::ComputeTag {
+  using argument_tags = tmpl::list<
+      Tags::InitialGaugeH<SpatialDim, Frame>, ::gr::Tags::Lapse<DataVector>,
+      ::gr::Tags::Shift<SpatialDim, Frame, DataVector>,
+      ::gr::Tags::SqrtDetSpatialMetric<DataVector>,
+      ::gr::Tags::SpacetimeMetric<SpatialDim, Frame, DataVector>, ::Tags::Time,
+      OptionTags::GaugeHRollOnStartTime, OptionTags::GaugeHRollOnTimeWindow,
+      ::Tags::Coordinates<SpatialDim, Frame>,
+      OptionTags::GaugeHSpatialWeightDecayWidth<Frame>>;
+
+  static typename db::item_type<Tags::GaugeH<SpatialDim, Frame>> function(
+      const typename db::item_type<Tags::InitialGaugeH<SpatialDim, Frame>>&
+          gauge_h_init,
+      const Scalar<DataVector>& lapse,
+      const tnsr::I<DataVector, SpatialDim, Frame>& shift,
+      const Scalar<DataVector>& sqrt_det_spatial_metric,
+      const tnsr::aa<DataVector, SpatialDim, Frame>& spacetime_metric,
+      const Time& time, const double& t_start, const double& sigma_t,
+      const tnsr::I<DataVector, SpatialDim, Frame>& coords,
+      const double& sigma_r) noexcept;
+};
+
+/*!
+ * \brief Damped harmonic gauge source function.
+ *
+ * \details Computes the gauge source function designed for binary black hole
+ * evolutions. These have been taken from \cite Szilagyi2009qz and
+ * \cite Deppe2018uye.
+ *
+ * The covariant form of the source function \f$H_a\f$ is written as:
+ *
+ * \f{align*}
+ * H_a :=  [1 - R_{H_\mathrm{init}}(t)] H_a^\mathrm{init} +
+ *  [\mu_{L1} \mathrm{log}(\sqrt{g}/N) + \mu_{L2} \mathrm{log}(1/N)] t_a
+ *   - \mu_S g_{ai} N^i / N
+ * \f}
+ *
+ * where \f$N, N^k\f$ are the lapse and shift respectively,
+ * \f$t_a\f$ is the unit normal one-form to the spatial slice,
+ *
+ * \f{align*}
+ *  \mu_{L1} &= A_{L1} R_{L1}(t) W(x^i) \mathrm{log}(\sqrt{g}/N)^{e_{L1}}, \\
+ *  \mu_{L2} &= A_{L2} R_{L2}(t) W(x^i) \mathrm{log}(1/N)^{e_{L2}}, \\
+ *  \mu_{S} &= A_{S} R_{S}(t) W(x^i) \mathrm{log}(\sqrt{g}/N)^{e_{S}},
+ * \f}
+ *
+ * temporal roll-on functions \f$ R_X(t)\f$ (with label \f$ X\f$) are:
+ *
+ * \f{align*}
+ * \begin{array}{ll}
+ *     R_X(t) & = 1 - \exp[-((t - t_{0,X})/ \sigma_t^X)^4], & t\geq t_{0,X} \\
+ *     & = 0, & t< t_{0,X} \\
+ * \end{array}
+ * \f}
+ *
```

Comment:
I have raised an issue for this (#1516) though, and assigned it to myself. Once this code is in develop, I can make any remaining changes required for SpECTRE's GH gauge to reproduce SpEC's. A couple of questions though:

> The time rolloff options here cannot reproduce the default behavior in SpEC for BBHs. This is not necessarily a bad thing (the SpEC time rolloff functions are unnecessarily complicated and I think the ones here are cleaner), but if we are to do comparisons of SpECTRE vs SpEC there will need to be some further code changes in one of the two codes to ensure both are in the same gauge.
> 
Isnt time rolloff in SpEC controlled with `RolloffFormula` [here](https://github.com/sxs-collaboration/spec/blob/4f6b076f504533de4ad1d4520bb49e2b131ce7a4/EvolutionSystems/GeneralizedHarmonic/DampedHarmonicGaugeItems.hpp#L219)? I ask because that is what has been implemented in [here](https://github.com/sxs-collaboration/spectre/blob/27d25e42c8871c26278f04c8dd4b2eaddd8b9725/src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp#L94) and [here](https://github.com/sxs-collaboration/spectre/blob/27d25e42c8871c26278f04c8dd4b2eaddd8b9725/src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp#L107).

> Furthermore, the spatial rolloffs are not the same in SpECTRE as in SpEC BBHs (SpEC uses a power of 2, not 4).
> 
But, spatial rolloff here uses a power of 2 only (not 4), e.g. see [here](https://github.com/sxs-collaboration/spectre/blob/27d25e42c8871c26278f04c8dd4b2eaddd8b9725/src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp#L35).

Reply:
For SpEC, I think the time rolloff is the same as SpECTRE, but the time rollon is the product of two generalized gaussians, i.e. `R_{L1}(t)_{SpEC} = R_{H_\mathrm{init}}(t)_{SpECTRE} R_{L1}(t)_{SpECTRE}` and similar for R_L2 and R_S.  But please correct me if I am wrong.

My bad on the spatial rolloff; yes I agree both use a power of 2.

Reply:
> For SpEC, I think the time rolloff is the same as SpECTRE, but the time rollon is the product of two generalized gaussians, i.e. `R_{L1}(t)_{SpEC} = R_{H_\mathrm{init}}(t)_{SpECTRE} R_{L1}(t)_{SpECTRE}` and similar for R_L2 and R_S. 

yes, `R_{L1}(t)_{SpEC} = R_{H_\mathrm{init}}(t)_{SpECTRE} R_{L1}(t)_{SpECTRE}` is about right. Why would one choose two generalized Gaussians for roll-on is not very clear to me though :)